### PR TITLE
Added Display Of Social Medias And Modifying Them

### PR DESCRIPTION
**Pull request by GuenoleCHERUBIN**
Added the display of social medias in a view
And the possibility to modify a selected social media, 
on an other view

### CODE REVIEW

File: controllers/ManagementController.php [hunk 1]
```diff
@@ -196,5 +198,77 @@ private function createAccount(CreateAccountForm $form)
...
+    }
+
+    /**
+     * @param string $id
+     * @return string
+     * @throws CannotSaveException
+     */
+    private function actionModifySocialMedias(string $id)
+    {
+        $socialMedia = $this->checkIfSocialMediaExist($id);
```
Comment (EmileCalixte) [inline controllers/ManagementController.php]: checkIfSocialMediaExists*

File: controllers/ManagementController.php [hunk 2]
```diff
@@ -196,5 +198,77 @@ private function createAccount(CreateAccountForm $form)
             throw new CannotSaveException($user);
         }
     }
+
+    /**
```
Comment (alexisljn) [inline controllers/ManagementController.php:203]: you should explain what method do

File: controllers/ManagementController.php [hunk 3]
```diff
@@ -196,5 +198,77 @@ private function createAccount(CreateAccountForm $form)
...
+            'pagination' => [
+                'pageSize' => 10,
+            ],
+        ]);
+        return $this->render('social-medias', [
+            'socialMediasDataProvider' => $socialMediasDataProvider
+        ]);
+    }
+
+    /**
```
Comment (alexisljn) [inline controllers/ManagementController.php:226]: you should explain what method do

File: controllers/ManagementController.php [hunk 4]
```diff
@@ -196,5 +198,77 @@ private function createAccount(CreateAccountForm $form)
...
+
+
+
+        return $this->render('modify-social-medias', [
+            'formModifySocialMediasModel' => $formModifySocialMedias,
+            'socialMedia' => $socialMedia
+        ]);
+    }
+
+    /**
```
Comment (alexisljn) [inline controllers/ManagementController.php:252]: you should explain what method do

File: controllers/ManagementController.php [hunk 5]
```diff
@@ -196,5 +198,77 @@ private function createAccount(CreateAccountForm $form)
...
+    private function checkIfSocialMediaExist(string $id)
+    {
+        $unauthorizedException = NotFoundHttpException::class;
+        if (!is_null($socialMedia = EnabledSocialMedia::findOne(['social_media_name' => $id]))) {
+            return $socialMedia;
+        }
+        throw new $unauthorizedException();
+    }
+
+    /**
```
Comment (alexisljn) [inline controllers/ManagementController.php:267]: you should explain what method do

**Review by alexisljn — CHANGES_REQUESTED** [review]

File: controllers/ManagementController.php [hunk 6]
```diff
@@ -196,5 +198,77 @@ private function createAccount(CreateAccountForm $form)
...
+        ]);
+    }
+
+    /**
+     * @param string $id
+     * @return EnabledSocialMedia|null
+     */
+    private function checkIfSocialMediaExist(string $id)
+    {
+        $unauthorizedException = NotFoundHttpException::class;
```
Comment (EmileCalixte) [inline controllers/ManagementController.php]: Why is it called unauthorizedException ?

File: controllers/ManagementController.php [hunk 7]
```diff
@@ -196,5 +198,77 @@ private function createAccount(CreateAccountForm $form)
...
+    }
+
+    /**
+     * @param ModifySocialMediaInformationsForm $formModifySocialMedias
+     * @param $socialMedia
+     * @throws CannotSaveException
+     */
+    private function updateSocialMedia(ModifySocialMediaInformationsForm $formModifySocialMedias, $socialMedia)
+    {
+           $socialMedia->is_enabled = $formModifySocialMedias->is_enabled;
```
Comment (EmileCalixte) [inline controllers/ManagementController.php]: I think there is an indentation error

File: models/databaseModels/EnabledSocialMedia.php [hunk 8]
```diff
@@ -3,14 +3,15 @@
 namespace app\models\databaseModels;
 
 use Yii;
+use yii\db\ActiveRecord;
```
Comment (EmileCalixte) [inline models/databaseModels/EnabledSocialMedia.php]: Never edit this file, it's auto-generated by the Yii code generator

**Review by EmileCalixte — CHANGES_REQUESTED** [review]

**Review by EmileCalixte — APPROVED** [review]

**Review by alexisljn — APPROVED** [review]
Nice !

**Pull request merged**